selfie-camera/src/main/java/org/odk/collect/selfiecamera/CaptureSelfieActivity.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.odk.collect.selfiecamera
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.Window
+import android.view.WindowManager
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageCapture
+import androidx.camera.core.ImageCaptureException
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.video.FileOutputOptions
+import androidx.camera.video.Recorder
+import androidx.camera.video.Recording
+import androidx.camera.video.VideoCapture
+import androidx.camera.video.VideoRecordEvent
+import androidx.camera.view.PreviewView
+import androidx.core.content.ContextCompat
+import org.odk.collect.androidshared.ui.ToastUtils.showLongToast
+import org.odk.collect.externalapp.ExternalAppUtils
+import org.odk.collect.permissions.PermissionsChecker
+import org.odk.collect.strings.localization.LocalizedActivity
+import java.io.File
+
+class CaptureSelfieActivity : LocalizedActivity() {
+
+    private var recording: Recording? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (
+            !PermissionsChecker(this).isPermissionGranted(
+                Manifest.permission.CAMERA,
+                Manifest.permission.RECORD_AUDIO
+            )
+        ) {
+            finish()
+            return
+        }
+
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        window.setFlags(
+            WindowManager.LayoutParams.FLAG_FULLSCREEN,
+            WindowManager.LayoutParams.FLAG_FULLSCREEN
+        )
+        setContentView(R.layout.activity_capture_selfie)
+
+        val cameraProvider = ProcessCameraProvider.getInstance(this)
+        cameraProvider.addListener(
+            {
+                if (intent.getBooleanExtra(EXTRA_VIDEO, false)) {
+                    setupVideo(cameraProvider.get())
+                } else {
+                    setupStillImage(cameraProvider.get())
+                }
+            },
+            ContextCompat.getMainExecutor(this)
+        )
+
+        showLongToast(this, R.string.take_picture_instruction)
```

Comment:
So we have the same text displayed for both image and video modes? This is `Tap the screen to take a picture` and if so then we should improve it.